packages/auto-instrumentations-node/src/instrumentations/extentions/http.ts

```diff
@@ -132,13 +132,15 @@ function createHttpResponseHook(
     if (response instanceof ServerResponse) {
       const originalWrite = response.write;
       response.write = function (chunk: any, callback) {
-        response.write = originalWrite;
         bodyHandler.addChunk(chunk);
         return originalWrite.call(this, chunk, callback);
       };
 
       const originalEnd = response.end;
       response.end = function (..._args: ResponseEndArgs) {
+        //return the 'write()' function to be the originalOne
+        //only after the end() funciton is called.
+        response.write = originalWrite;
         response.end = originalEnd;
```

Comment:
I couldn't understand why this is required

Reply:
In case the user will call write() *after* he calls *end() - which he shouldn't, according to the documentation - we don't want our hook to be called.